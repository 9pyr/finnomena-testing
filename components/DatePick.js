import styles from 'styles/DatePicker.module.scss'

import React from 'react'
import { DatePicker } from 'antd'

const { RangePicker } = DatePicker
const DatePick = ({ value, onChange }) => {
  return (
    <div className={styles.date_picker}>
      <RangePicker value={value} onChange={onChange} />
    </div>
  )
}

export default DatePick
