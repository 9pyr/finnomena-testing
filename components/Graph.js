import styles from 'styles/Graph.module.scss'

import React from 'react'
import { Area } from '@ant-design/plots'

const Graph = ({ data = [] }) => {
  return (
    <div className={styles.graph_styled}>
      <Area
        renderer='svg'
        height={200}
        autoFit={false}
        data={data}
        xField={'thailand_fund_code'}
        yField={'nav_return'}
        xAxis={{ label: null, line: null }}
        yAxis={{ label: null, grid: null }}
        tooltip={{ customContent }}
        smooth={true}
        areaStyle={() => ({
          fill: 'l(270) 0:#ffffff 0.5:#7ec2f3 1:#1890ff',
        })}
      />
    </div>
  )
}

export default Graph

export const customContent = (title, data) => {
  const { value } = data?.[0] || {}
  return (
    <div className={styles.tooltip_custom}>
      <div>
        ชื่อกองทุน: <span className={styles.title}>{title}</span>
      </div>
      <div>
        ผลตอบแทน: <span className={styles.text_highlight}>{value}</span>
      </div>
    </div>
  )
}
