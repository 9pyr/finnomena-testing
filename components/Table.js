import React from 'react'
import dayjs from 'dayjs'
import { Table as AntdTable } from 'antd'

const Table = ({ dataSource, onChange }) => {
  return <AntdTable showSorterTooltip={false} columns={columns} dataSource={dataSource} onChange={onChange} rowKey={(record) => record.mstar_id} />
}

export default Table

const columns = [
  {
    title: 'ชื่อกองทุน',
    dataIndex: 'thailand_fund_code',
    key: 'thailand_fund_code',
    ellipsis: true,
    width: '30%',
  },
  {
    title: 'อันดับของกองทุน',
    dataIndex: 'nav_index',
    key: 'nav_index',
    sorter: (a, b) => a.nav_index - b.nav_index,
    ellipsis: true,
    width: '15%',
    align: 'center',
  },
  {
    title: 'เวลาที่ข้อมูลถูกอัพเดต',
    dataIndex: 'nav_date',
    key: 'nav_date',
    sorter: (a, b) => new Date(a.nav_date).getTime() - new Date(b.nav_date).getTime(),
    ellipsis: true,
    render: (text) => dayjs(text).format('DD MMMM YYYY'),
    width: '25%',
    align: 'center',
  },
  {
    title: 'ผลตอบแทน',
    dataIndex: 'nav_return',
    key: 'nav_return',
    sorter: (a, b) => a.nav_return - b.nav_return,
    ellipsis: true,
    width: '15%',
    align: 'right',
  },
  {
    title: 'ราคา',
    dataIndex: 'nav',
    key: 'nav',
    sorter: (a, b) => a.nav - b.nav,
    ellipsis: true,
    width: '15%',
    align: 'right',
  },
]
