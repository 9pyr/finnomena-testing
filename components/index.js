export { default as Table } from './Table'
export { default as Graph } from './Graph'
export { default as DatePick } from './DatePick'