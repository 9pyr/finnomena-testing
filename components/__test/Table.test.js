import React from 'react'
import dayjs from 'dayjs'
import { Table } from 'components'
import { render } from '@testing-library/react'

const defaultProps = {
  dataSource: [
    {
      mstar_id: 'F00000ZMXD',
      thailand_fund_code: 'PRINCIPAL VNEQ-A',
      nav_return: 74.80117,
      nav: 14.154,
      nav_date: '2021-09-07T00:00:00.000Z',
      avg_return: 63.633823,
      nav_index: 1,
    },
    {
      mstar_id: 'F0000109C9',
      thailand_fund_code: 'PRINCIPAL VNEQ-I',
      nav_return: 74.79997,
      nav: 14.2656,
      nav_date: '2021-09-07T00:00:00.000Z',
      avg_return: 63.633823,
      nav_index: 2,
    },
    {
      mstar_id: 'F00000HHFC',
      thailand_fund_code: 'KT-OIL',
      nav_return: 73.33466,
      nav: 3.4868,
      nav_date: '2021-09-03T00:00:00.000Z',
      avg_return: 65.394614,
      nav_index: 3,
    },
    {
      mstar_id: 'F00000IT9T',
      thailand_fund_code: 'TMBOIL',
      nav_return: 71.99969,
      nav: 4.354,
      nav_date: '2021-09-06T00:00:00.000Z',
      avg_return: 65.394614,
      nav_index: 4,
    },
  ],
  onChange: () => {},
}
const setupValues = (props) => ({ ...defaultProps, ...props })

describe('Table Component', () => {
  test('render has props', () => {
    const props = setupValues()
    const { container } = render(<Table {...props} />)

    const antTableRow = container.getElementsByClassName('ant-table-row')

    expect(container).toMatchSnapshot()
    expect(antTableRow.length).toEqual(props.dataSource.length)
    props.dataSource.map((item, i) => {
      const antTableCell = antTableRow[i].getElementsByClassName('ant-table-cell')

      expect(antTableCell[0].textContent).toEqual(item.thailand_fund_code)
      expect(antTableCell[1].textContent).toEqual(String(item.nav_index))
      expect(antTableCell[2].textContent).toEqual(dayjs(item.nav_date).format('DD MMMM YYYY'))
      expect(antTableCell[3].textContent).toEqual(String(item.nav_return))
      expect(antTableCell[4].textContent).toEqual(String(item.nav))
    })
  })

  test('render is props undefined', () => {
    const { container } = render(<Table />)

    const antTableRow = container.getElementsByClassName('ant-table-row')

    expect(container).toMatchSnapshot()
    expect(antTableRow.length).toEqual(0)
  })
})
