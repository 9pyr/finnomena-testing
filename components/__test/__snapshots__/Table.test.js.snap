// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component render has props 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: fixed;"
              >
                <colgroup>
                  <col
                    style="width: 30%;"
                  />
                  <col
                    style="width: 15%;"
                  />
                  <col
                    style="width: 25%;"
                  />
                  <col
                    style="width: 15%;"
                  />
                  <col
                    style="width: 15%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis"
                      title="ชื่อกองทุน"
                    >
                      ชื่อกองทุน
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          อันดับของกองทุน
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          เวลาที่ข้อมูลถูกอัพเดต
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: right;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          ผลตอบแทน
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: right;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          ราคา
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="F00000ZMXD"
                  >
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      title="PRINCIPAL VNEQ-A"
                    >
                      PRINCIPAL VNEQ-A
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="1"
                    >
                      1
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="07 September 2021"
                    >
                      07 September 2021
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="74.80117"
                    >
                      74.80117
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="14.154"
                    >
                      14.154
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="F0000109C9"
                  >
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      title="PRINCIPAL VNEQ-I"
                    >
                      PRINCIPAL VNEQ-I
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="2"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="07 September 2021"
                    >
                      07 September 2021
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="74.79997"
                    >
                      74.79997
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="14.2656"
                    >
                      14.2656
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="F00000HHFC"
                  >
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      title="KT-OIL"
                    >
                      KT-OIL
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="3"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="03 September 2021"
                    >
                      03 September 2021
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="73.33466"
                    >
                      73.33466
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="3.4868"
                    >
                      3.4868
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="F00000IT9T"
                  >
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      title="TMBOIL"
                    >
                      TMBOIL
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="4"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: center;"
                      title="06 September 2021"
                    >
                      06 September 2021
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="71.99969"
                    >
                      71.99969
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-ellipsis"
                      style="text-align: right;"
                      title="4.354"
                    >
                      4.354
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination ant-table-pagination-right"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-prev ant-pagination-disabled"
            title="Previous Page"
          >
            <button
              class="ant-pagination-item-link"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="left"
                class="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a
              rel="nofollow"
            >
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-next ant-pagination-disabled"
            title="Next Page"
          >
            <button
              class="ant-pagination-item-link"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table Component render is props undefined 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-empty"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: fixed;"
              >
                <colgroup>
                  <col
                    style="width: 30%;"
                  />
                  <col
                    style="width: 15%;"
                  />
                  <col
                    style="width: 25%;"
                  />
                  <col
                    style="width: 15%;"
                  />
                  <col
                    style="width: 15%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis"
                      title="ชื่อกองทุน"
                    >
                      ชื่อกองทุน
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          อันดับของกองทุน
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          เวลาที่ข้อมูลถูกอัพเดต
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: right;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          ผลตอบแทน
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters"
                      style="text-align: right;"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          ราคา
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-placeholder"
                  >
                    <td
                      class="ant-table-cell"
                      colspan="5"
                    >
                      <div
                        class="ant-empty ant-empty-normal"
                      >
                        <div
                          class="ant-empty-image"
                        >
                          <svg
                            class="ant-empty-img-simple"
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fill-rule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                class="ant-empty-img-simple-ellipse"
                                cx="32"
                                cy="33"
                                rx="32"
                                ry="7"
                              />
                              <g
                                class="ant-empty-img-simple-g"
                                fill-rule="nonzero"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  class="ant-empty-img-simple-path"
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <div
                          class="ant-empty-description"
                        >
                          No Data
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
